File: VirtualGardens/VirtualGardens.Services/Database/Zaposlenici.cs
using System;
using System.Collections.Generic;
using VirtualGardens.Services.BaseInterfaces;

namespace VirtualGardens.Services.Database;

public partial class Zaposlenici : ISoftDeletable
{
    public int ZaposlenikId { get; set; }

    public string Email { get; set; } = null!;

    public string Ime { get; set; } = null!;

    public string Prezime { get; set; } = null!;

    public string? BrojTelefona { get; set; }

    public string? Adresa { get; set; }

    public string? Grad { get; set; }

    public string? Drzava { get; set; }

    public bool JeAktivan { get; set; } = true;

    public DateTime? DatumRodjenja { get; set; }

    public virtual ICollection<Nalozi> Nalozis { get; set; } = new List<Nalozi>();
    public bool IsDeleted { get; set; } = false;
    public DateTime? VrijemeBrisanja { get; set; }
}

File: UI/virtualgardens_mobile/lib/models/zaposlenici.dart
import 'package:json_annotation/json_annotation.dart';

part 'zaposlenici.g.dart';

@JsonSerializable()
class Zaposlenik {
  int zaposlenikId;
  String email;
  String ime;
  String prezime;
  String? brojTelefona;
  String? adresa;
  String? grad;
  String? drzava;
  bool jeAktivan;
  DateTime? datumRodjenja;

  Zaposlenik({
    required this.zaposlenikId,
    required this.email,
    required this.ime,
    required this.prezime,
    this.brojTelefona,
    this.adresa,
    this.grad,
    this.drzava,
    required this.jeAktivan,
    this.datumRodjenja,
  });

  factory Zaposlenik.fromJson(Map<String, dynamic> json) =>
      _$ZaposlenikFromJson(json);

  Map<String, dynamic> toJson() => _$ZaposlenikToJson(this);
}
